workflow/run_LOH_app.cwl

```diff
@@ -3,7 +3,57 @@
+label: Kids First Loss of Heterozygosity
+doc: | 
+ # Kids First Loss of Heterozygosity (LOH)
+
+  ![data service logo](https://github.com/d3b-center/d3b-research-workflows/raw/master/doc/kfdrc-logo-sm.png)
+
+  The Kids First Loss of Heterozygosity (aka LOH) is a CWL workflow that assesses the loss of heterozygosity in the tumor for rare germline calls filtered by gnomad_3_1_1_AF_popmax (typically < 0.01) or when gnomad_3_1_1_AF_popmax is not defined. This workflow is designed to analyze LOH for family trios as well as multiple proband tumor samples. 
+
+  ### Description
+
+  The Kids First Loss of Heterozygosity application is divided into two tools: Germline tool and Tumor tool.
+
+  #### Docker
+  Dockerfile: https://github.com/d3b-center/bixtools/tree/master/LOH
+
+  #### Germline Tool
+
+  Germline tool filters germline annotations to retain variants based on gnomad_3_1_1_AF_popmax (typically < 0.01) or when gnomad_3_1_1_AF_popmax is not defined. It requires vcf file, proband sample id, ram as required inputs and peddy file as optional input which is required for family trios. It outputs variant information such as gene, chr, start, stop, ref/alt alleles, ref/alt allele depths, variant allele frequency and list of coordinates that will be an input to tumor tool.
+
+  #### Tumor Tool
+  Tumor tool search in paired proband tumor sample for aligned reads in the regions where rare variants from the germline tool exists and exact allele/reference count, allele/reference depth and calculate the variant allele frequency VAF. Tumor tool have the capability to search multiple tumor samples for proband and if applicable, parental and maternal tumor samples. To exact reads from the bam/cram files, this tool utilizes [bam-readcount](https://github.com/genome/bam-readcount) and wraps it with python script to shape the output in a tabular format. 
+
+  ### LOH Inputs
+  ```
+  Germline tool
+    # Required  
+    BS_ID: { doc: provide BS id for germline normal,type: string }
+    frequency: { doc: provide popmax cutoff for rare germline variants, type: 'float?', default: 0.01 }
+    # Optional
+    ram_germline: {  doc: Provide ram (in GB) based on the size of vcf,type: 'int?', default: 8}
+    # Required for family trios otherwise not required
+    peddy_file: { doc: provide ped file for the trio, type: 'File?' }
+  Tumor tool
+    # Required
+    participant_id: { doc: provide participant id for this run, type: string }
+    bamscrams: { doc: tumor input file in cram or bam format with their index file, type: 'File[]' , secondaryFiles: [ { pattern: ".crai", required: false }, { pattern: ".bai", required: false } ] }
+    reference: { doc: human reference in fasta format with index file, type: File,secondaryFiles: [ .fai ],"sbg:suggestedValue": {class: File, path: 60639014357c3a53540ca7a3, name: Homo_sapiens_assembly38.fasta} }
```

Comment:
One small thing I think to round this out would be to add also the secondary file as a suggested input, so:
`reference: { doc: human reference in fasta format with index file, type: File,secondaryFiles: [ .fai ],"sbg:suggestedValue": {class: File, path: 60639014357c3a53540ca7a3, name: Homo_sapiens_assembly38.fasta, secondaryFiles: [{class: File, path: 60639016357c3a53540ca7af, name: Homo_sapiens_assembly38.fasta.fai}]} }`
Just double check that I have the brackets properly placed

Reply:
this is a preprocessing workflow yes